2016/proposals/narayan-aditya-visualizer.md

```diff
@@ -79,13 +87,135 @@ number of API calls left and the time of the day.
 
    -- DonaldKnuth
 
-### Database - PostgreSQL
+### Database - Django ORM
 
-The Django app will use PostgreSQL or SQLite as its
-database engine in production and SQLite database engine during
-development to store the survey responses fetched by the
+The SurveyMonkey Django app will use five major tables
+to store the survey responses fetched by the
 SurveyMonkey Django application.
 
+* **Survey**: The *Survey* table will consists of surveys fetched
+   and auto-populated by the [get_survey_details] API endpoint.
+* **Respondent**: The *Respondent* table stores basic user details
+  of all respondents for a given survey.
+* **Question**: Each survey row in the *Survey* table consists of
+  a list of questions, ranging from but not limited to
+  * Single choice / Dropdowns
+  * Matrix choice
+  * Text Box
+* **Answer**: Each question in the *Question* table has certain
+  answers that the respondent can select or give as an input. For eg:
+  * Single choice / Dropdowns: These questions have a list of answers
+    to them.
+  * Matrix choice: These questions have a row and a column of answers
+    to them.
+  * Text Box: The answer to these questions is a string.
+* **Response**: Each row in the *Response* table associates a respondent
+  with the question that was asked and the answer that
+  the respondent selected or inputted.
+
+I propose four database schemas.
```

Comment:
Is this summary good enough? These arguments have been mentioned as PROS and CONS within each proposal.
>Proposal 1 and 2 attempt to store all the fields of the *Answer* table as SQL columns, while proposal 3 and 4 attempt to save the fields by serializing it to a dictionary and saving them in a `TextField` and an `HStoreField` respectively. Storing fields in SQL columns allow us to impose type check restrictions (like the data under the column `answer_id` must be an integer). We cannot enforce such a check when the data is stored as a serialized dict in a `TextField` or an `HStoreField`
Proposal 1 and 4 allow us to filter/query responses based on a particular answer to a specific question (in our use case, the question is *Please select the workshop?* or the custom variable). Proposal 2 and 3 cannot filter/query responses. We have to resort to saving the field on which we intend to query as a separate column in the *Response* table, which introduces a redundancy.